### PR TITLE
Add initial support for custom bar-id

### PR DESCRIPTION
**Pull request by mikkeloscar**
Sets a default bar-id on all bars. This can be changed with the `id` sub-command (once implemented).

### CODE REVIEW

File: sway/commands.c [hunk 1]
```diff
@@ -1124,9 +1124,26 @@ static struct cmd_results *cmd_bar(int argc, char **argv) {
 	bar->tray_padding = config->bar.tray_padding;
 	list_add(config->bars, bar);
 
+	// set bar id
+	int i, n;
+	int len = 5; // "bar-" + \0
```
Comment (SirCmpwn) [inline sway/commands.c]: Can you define this closer to where it's used, and add const?

Comment (mikkeloscar) [inline sway/commands.c]: I have moved it down to the do-while loop, but I'm not sure what should be const?

Comment (SirCmpwn) [inline sway/commands.c]: Nevermind, was me being stupid. I think there's a function in here somewhere that tells you how big a number is, we should reuse it.

Comment (SirCmpwn) [inline sway/commands.c]: https://github.com/SirCmpwn/sway/blob/master/swaygrab/main.c#L16-L24

Comment (SirCmpwn) [inline sway/commands.c]: Anyone want to take bets on when we'll get the first bug report for screens >1,000,000 pixels wide?

Comment (mikkeloscar) [inline sway/commands.c]: `numlen` is defined in `swaygrab/main.c`, where should I put it for general use?

Comment (SirCmpwn) [inline sway/commands.c]: `common/util.c` or something

**Comment by mikkeloscar** [comment]
I have moved `numlen()` to `common/util.c` and merged it with `sway/util.c` which only had one function.

I have also updated the bar-id stuff to use `numlen`.

Hope it's ok to have all this in a single PR.

**Comment by SirCmpwn** [comment]
It's fine for this to all be in one PR. Thanks for taking care of that.

**Pull request merged**